modules/library/main/src/test/java/org/geotools/filter/function/FilterFunction_inTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *    
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+package org.geotools.filter.function;
+
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.Function;
+import org.opengis.filter.expression.PropertyName;
+
+/**
+ * Test class for {@link FilterFunction_in}.
+ * 
+ * @author Stefano Costa, GeoSolutions
+ * 
+ * 
+ * 
+ * @source $URL$
+ *
+ */
+public class FilterFunction_inTest extends FunctionTestSupport {
+
+    private static final String FUNCTION_NAME = "in";
+
+    private SimpleFeature feature;
+
+    public FilterFunction_inTest(String testName) {
+        super(testName);
+    }
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+
+        feature = featureCollection.features().next();
+    }
+
+    public void testIntPresent() {
+        PropertyName exp = ff.property("foo");
+        Function func = ff
+                .function(FUNCTION_NAME, exp, ff.literal(3), ff.literal(4), ff.literal(5));
+        Object result = func.evaluate(feature);
+        assertEquals(true, result);
+    }
+
+    public void testIntMissing() {
+        PropertyName exp = ff.property("foo");
+        Function func = ff.function(FUNCTION_NAME, exp, ff.literal(1), ff.literal(2));
+        Object result = func.evaluate(feature);
+        assertEquals(false, result);
+    }
+
+    public void testDoublePresent() {
+        PropertyName exp = ff.property("bar");
+        Function func = ff.function(FUNCTION_NAME, exp, ff.literal(2.5), ff.literal(2.6),
+                ff.literal(3.0));
+        Object result = func.evaluate(feature);
+        assertEquals(true, result);
+    }
+
+    public void testDoubleMissing() {
+        PropertyName exp = ff.property("bar");
+        Function func = ff.function(FUNCTION_NAME, exp, ff.literal(2.499999), ff.literal(2.500001));
+        Object result = func.evaluate(feature);
+        assertEquals(false, result);
+    }
```

Comment:
To be clear, we need a test of:

Function func = ff.function(FUNCTION_NAME, exp, ff.literal(2.499999), ff.literal("2.500001"),ff.literal(2));